NewDawn/NewDawn/LocalStorageUtil.swift

```diff
@@ -95,24 +93,18 @@ class LoginUserUtil {
         if let user_id = getLoginUserId(), let accessToken = getAccessToken() {
             // Check if the user profile has already fetched and stored in local storage
             if let user_profile: UserProfile? = LocalStorageUtil.localReadKeyValueStruct(key: LoginUserUtil.LOGIN_USER_PROFILE) {
-                DispatchQueue.main.async {
-                    callback(user_profile)
-                }
+                callback(user_profile)
             } else {
                 LoginUserUtil.fetchUserProfile(user_id: user_id, accessToken: accessToken) {
                     user_profile in
-                    LocalStorageUtil.localStoreKeyValue(key: LoginUserUtil.LOGIN_USER_PROFILE, value: user_profile)
-                    DispatchQueue.main.async {
-                        callback(user_profile)
-                    }
+                    LocalStorageUtil.localStoreKeyValueStruct(key: LoginUserUtil.LOGIN_USER_PROFILE, value: user_profile)
```

Comment:
@duckmoll Here I shouldn't use localStoreKeyValue since the value is a struct.
When I commented out this line, or change it to the correct one, the app stop crashing immediately.

Reply:
I see.

Reply:
Yes.